src/UI/Factory.php

```diff
@@ -22,10 +22,14 @@
 	 *     They indicate the functionality or nature of a text-element or context:
 	 *     Icons will mainly be used in front of object-titles, e.g. in the
 	 *     header, the tree and in repository listing.
+	 *     Icons can be disabled. Disabled Icons visually communicate that they
+	 *     can not be worked with by users, i.e. a precondition is not met yet or
+	 *     status is offline. Technically this means that read access is not granted.
```

Comment:
Please remove that `Technically...` part and just add that as another example why an icon possibly cannot be used.

Reply:
Should they even be displayed in the given example? Can you elaborate a little on this?

Reply:
` i.e. a precondition is not met yet, status is offline or read access is not granted.`

Reply:
@klees ?

Probably my point was not clear enough:
Please be more specific in which case the disabled icon should be displayed and in which case the object is not displayed at all. I can only guess, that this is intended to be used if somebody has read access but others might not, correct? E.g. some admin would see this and this indicates that others might not (as the "offline" label currently does), correct? In this case others would not see the object at all, and the admin gets a hint of the state of this object. Or is it, somebody has "visible" rights but not "read"? Or both?

This differentiation seems important to me, since the first indicates the status of something for the user configuring it, but not the state of access for the current user itself, while the second directly communicates the options for the user currently staring at the icon.

Reply:
objects and their icons are usually shown are shown, if visible permission is given. For these visible objects the disabled icon should indicate if "read access" is given or not (not taking any other permissions into account). Read access may not be given if:
- no read permission is given
- no precondition is met
- object is offline
-... (anything that limits read access in objects access classes)

Reply:
Ok, I thought your remark was for me. I think that my opinion here is exactly opposite. I would not want to discuss, solve or document these questions in the context of the UI-framework. For me `Disabled Icons visually communicate that they can not be worked with by users` with some examples is absolutely enough.

Reply:
"Disabled Icons visually communicate that they can not be worked with by users" is ok for me too. This was just a technical explanation. We add links to the concrete FRs.

Reply:
Hmm, somehow i still believe that my point is not yet understood.

> For me Disabled Icons visually communicate that they can not be worked with by users with some examples is absolutely enough.

@klees I disagree. 

>Disabled Icons visually communicate that they can not be worked with by users

You see that there are two different options into which this could be understood? It could mean that there users out there which have no access (cause this is offline or similar), and this is shown to someone that might have all possible permissions. Or this can always ONLY directly relate to the permission of the user staring at it. From what I gather from @alex40724  this is the case. If we introduce state on icons, we need to agree on the meaning of those states. 

For me some wording as:
"The disabled is always to be understood in the context of the current user and not some global state such as offline. E.g. if a user has the permission to read and write the content of an offline object, then the icon shown to this user for this object should not be marked as disabled."

would be helpful.

However, if it is clear to anybody else, I can live without.

Reply:
> You see that there are two different options into which this could be understood?

Yes, now I see =) Your proposed wording is fine for me. I'm 100% sure if that distinction is really required, though.

Reply:
Oliver is proposing ti show the disabled icon for users having write permission, if a portfolio is not shared/offline. Even, if the user is the author:
https://www.ilias.de/docu/goto_docu_wiki_wpage_5052_1357.html

Reply:
@alex40724 Thanks a lot. Now we are at the core of the question: Is this fine? Can the disabled icon be used both ways: To mark something as not accessible for the current user AND/OR to mark something as not accessible for other users? I could live with that, but would like it to be clearly pointed out somehow.